src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/Package/Update/PackageUpdateCommandRunner.cs

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Commands;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Credentials;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Commands.Package.Update
+{
+    internal static class PackageUpdateCommandRunner
+    {
+        internal static async Task<int> Run(PackageUpdateArgs args, ILoggerWithColor logger, IDGSpecFactory dGSpecFactory, MSBuildAPIUtility msbuild, CancellationToken cancellationToken)
+        {
+            XPlatUtility.ConfigureProtocol();
+            DefaultCredentialServiceUtility.SetupDefaultCredentialService(logger, nonInteractive: false);
+
+            // 1. Get DGSpec for project/solution
+            // 2. Find suitable version of package(s) to update
+            // 3. Preview restore to validate changes
+            // 4. Update MSBuild files
+            // 5. Commit restore if everything successful
+
+            // 1. Get DGSpec for project/solution
```

Comment:
```suggestion
```

Reply:
My intention was to show the overall algorithm at the top of the method, and then each "step" has the step number & name repeated above the code that implements it, to make it clear which part of the algorithm the code is relevant to.